plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererUserModificationTracker.kt

```diff
@@ -152,28 +156,18 @@ class CodeWhispererUserModificationTracker(private val project: Project) : Dispo
 
         if (file == null || (!file.isValid)) {
             sendModificationTelemetry(acceptedSuggestion, null)
-            // temp remove event sent as further discussion needed for metric calculation
-            // sendUserModificationTelemetryToServiceAPI(acceptedSuggestion, 1.0)
+            sendUserModificationTelemetryToServiceAPI(acceptedSuggestion)
         } else {
-            try {
-                /**
-                 * this try-catch is to check if the offsets are valid since the method does not return null
-                 */
-                val document = runReadAction {
-                    FileDocumentManager.getInstance().getDocument(file)
-                }
-                val currentString = document?.getText(
-                    TextRange(acceptedSuggestion.range.startOffset, acceptedSuggestion.range.endOffset)
-                )
-                val modificationPercentage = checkDiff(currentString?.trim(), acceptedSuggestion.suggestion.trim())
-                sendModificationTelemetry(acceptedSuggestion, modificationPercentage)
-                // temp remove event sent as further discussion needed for metric calculation
-                // sendUserModificationTelemetryToServiceAPI(acceptedSuggestion, modificationPercentage)
-            } catch (e: Exception) {
-                sendModificationTelemetry(acceptedSuggestion, null)
-                // temp remove event sent as further discussion needed for metric calculation
-                // sendUserModificationTelemetryToServiceAPI(acceptedSuggestion, 1.0)
+            // Will remove this later when we truly don't need toolkit user modification telemetry anymore
+            val document = runReadAction {
+                FileDocumentManager.getInstance().getDocument(file)
```

Comment:
we can do this later, for simplicity, we should simply add document as a field of AcceptedSuggestionEntry. https://github.com/aws/aws-toolkit-jetbrains/blob/main/plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorManager.kt#L56

Reply:
ok